### ISSUE TITLE
Issue about model parameters

### ISSUE THREAD

**Issue by Kelly510** [ChelsieLei/EZ-HOI#9]
Hi! The [txtcls_ctx_pt](https://github.com/ChelsieLei/EZ-HOI/blob/397b93454a20d717a84a0a3abe8e142e470a1aca/upt_tip_cache_model_free_finetune_distillself.py#L529) is defined as a list of nn.Parameters, which is not saved in the checkpoint. I suppose this is a critical problem as the txtcls_ctx_pt is always randomly initialized even when inference?

**Comment by ChelsieLei** [comment]
Hi,

We use a list of nn.Parameter() to initialize the learnable prompts, as shown [here](https://github.com/ChelsieLei/EZ-HOI/blob/main/upt_tip_cache_model_free_finetune_distillself.py#L531) instead of nn.Parameters(), which will be saved as part of the model parameters in the checkpoints. Thus, it won’t be randomly initialized in inference.

**Comment by Kelly510** [comment]
I searched the checkpoint files such as hico_ezhoi_uv_vitbase.pt, but found no txtcls_ctx_pt. I think the nn.Module only saves its nn.Parameter. But when txtcls_ctx_pt is recognized as a list, the model does not save it as its checkpoint automatically. Please see the example below. 

![Image](https://github.com/user-attachments/assets/892e24c2-cb57-4b92-9708-7bdf706a7547)

Or is there other place to deal with this especially?

**Comment by ChelsieLei** [comment]
I see. I think a more precise way for the initialization should be
```
self.txtcls_ctx_pt = nn.ParameterList([nn.Parameter(torch.randn(n_ctx, ctx_dim))   for _ in range(self.compound_prompts_depth)])
```

Thank you to bring this out. I will correct the code.
However, I think this issue does not influence the results too much, because when we ran the inference code with our provided pre-trained checkpoints several times, the results keep similar and consistent. 
But I will  correct the code and also provide new checkpoints with  self.txtcls_ctx_pt stored in checkpoints later.

**Comment by Kelly510** [comment]
Thank you very much! 
Maybe the evaluation results are consistent because self.txtcls_ctx_pt is initialized with the same random seed. Hopefully the performance of this method will be better when self.txtcls_ctx_pt is well loaded from the trained checkpoints!

**Comment by Kelly510** [comment]
Hi Chelsie, 

I have corrected the code and retrained the model on HICO-DET under the unseen_verb setting. The updated results are as follows:

![Image](https://github.com/user-attachments/assets/89a9daad-9b9b-44dd-9407-5558ef0cc56e)

Contrary to my expectations, the performance does not improve when self.txtcls_ctx_pt is trained. Could you help explain this phenomenon?

**Comment by ChelsieLei** [comment]
Hi, maybe this issue doesn't influence the performance too much. 
Also, maybe you can conduct the original version experiments on your devices for the ablation study. Different devices may lead to a bit different performances.
